builtin/providers/azurerm/resource_arm_virtual_network.go

```diff
@@ -257,16 +278,55 @@ func getVirtualNetworkProperties(d *schema.ResourceData) *network.VirtualNetwork
 			DNSServers: &dnses,
 		},
 		Subnets: &subnets,
-	}
+	}, nil
```

Comment:
given we're returning two objects here, IMO this would read cleaner if we assigned this to a variable and then returned those, e.g.:

```
properties := &network....
return properties, nil
```